### ISSUE TITLE
Cannot Install: SHA256 mismatch when installing with Homebrew

### ISSUE THREAD

**Issue by danielyli** [casperjs/casperjs#1799]
The checksum for CasperJS 1.1.4 is incorrect in Homebrew, preventing installation:

```
$ brew install casperjs
==> Using the sandbox
==> Downloading https://github.com/casperjs/casperjs/archive/1.1.4.tar.gz
==> Downloading from https://codeload.github.com/casperjs/casperjs/tar.gz/1.1.4
######################################################################## 100.0%
Error: SHA256 mismatch
Expected: 3afa9210a2676cd9f5cda6227af0b656655b796107108562ec1fef550bdb11e7
Actual: b165f70943c44c164c55a022593bea51cb37cd87e5999ac7acacf14a31e5cc6c
Archive: /Users/d/Library/Caches/Homebrew/casperjs-1.1.4.tar.gz
To retry an incomplete download, remove the file above.
```

**Comment by danielyli** [comment]
It looks like @ilovezfs signed off on the relevant commit:

https://github.com/Homebrew/homebrew-core/commit/6f18d93cf28b831ec0bad9e04042b460859f928f

**Comment by jefleponot** [comment]
Ho @danielyli 

I don't know how to change it. I have never used the "brew" command.

**Comment by ilovezfs** [comment]
brew itself isn't relevant. Why did the checksum change?

**Comment by jefleponot** [comment]
I have recreated the package 1.1.4 ... so it could be the cause

**Comment by jefleponot** [comment]
Yes

**Comment by ilovezfs** [comment]
@jefleponot In future: please don't retag Git-based projects for a few reasons:

- when you retag you break the from-source build for anyone using Homebrew (and also possibly other projects)
- when you retag we don't know until a PR is filed whether it was intentional or you've been hacked
- we (and you) now can't easily differentiate between users using the old tag or the new one as they have the same Homebrew version number
- when you retag anyone who has done `git fetch --tags` and runs it again will not receive an updated tag. This is because Git is designed to not update tags as they are meant to be permanent. Retagging in Git-land is equivalent to force-pushing to a public branch (i.e. bad)

As a result of all this the ideal thing to do in this case is a `1.1.5` or `1.1.4.1` release. Thanks!

**Comment by jefleponot** [comment]
Thanks for all your explanation.
I will create a new tag soon

**Comment by danielyli** [comment]
FYI, the checksum is still incorrect as of 3/31 6 PM Pacific.

I'm curious—does this reveal a larger security issue, namely that someone could change the tarball without authorization?  Or does only @ilovezfs have permissions to update the checksum as well?

**Comment by ilovezfs** [comment]
The only people who can update the checksums are https://github.com/Homebrew/brew#who-are-you and we require 2FA for the organization.

>How easy is it for someone to spoof the payload and provide a correct checksum?

If you crack https and SHA-256, we have a problem.

**Comment by danielyli** [comment]
> The only people who can update the checksums are https://github.com/Homebrew/brew#who-are-you and we require 2FA for the organization.

Excellent, thank you.

**Comment by danielyli** [comment]
@jefleponot, any updates on the new tag?

**Comment by danielyli** [comment]
@ilovezfs Is it possible to update the Homebrew formula for 1.1.4 as a one-time thing?

**Comment by danielyli** [comment]
@ilovezfs Bump—is it possible to update?  This formula has been unusable for a few weeks.

**Comment by jefleponot** [comment]
i will create a 1.1.4.1 THAT WILL SOLVE THE BUG

**Comment by danielyli** [comment]
Thank you, @jefleponot, for promising to create a 1.1.4.1 tag to solve the issue.

If I understood @ilovezfs correctly, I believe the bug is simply that the original 1.1.4 tag was moved, and the solution is just to create a new 1.1.4.1 tag that points to the current 1.1.4 commit.  I don't see how the solution could take more than a minute to implement.  Why is it taking so long?

**Comment by jefleponot** [comment]
Hi @danielyli,

We have trouble with the casperJS.exe #1800. So We plan to commit a fix and create the new version 1.1.4.1 at the same time.

Regards

**Comment by jefleponot** [comment]
@danielyli 

Could you please now re-test with version 1.1.4-1 ?

Regards

**Comment by ilovezfs** [comment]
I've shipped 1.1.4-1 in Homebrew: https://github.com/Homebrew/homebrew-core/pull/12865

**Comment by danielyli** [comment]
Confirmed.  Installs with Homebrew.

Thanks for all your work, @jefleponot and @ilovezfs!
